common/Pid.cpp

```diff
@@ -35,16 +35,19 @@ Pid::~Pid()
 
 void Pid::setWindup(unsigned int w)
 {
-	if (w > 0)
+	if (w != _windup) 
 	{
-		if (w != _windup) {
-			_windup = w;
-			_oldErr = new float[_windup];
-			memset(_oldErr, 0, sizeof(float)*_windup);
+		if (_oldErr) 
+			delete[] _oldErr;
```

Comment:
You have to make sure to set `_oldErr` to nullptr after you deallocate it, otherwise it points to invalid memory.  There are places, like the destructor, that try to access or delete it if it's not nullptr.